src/frontend/src/binder/expr/column.rs

```diff
@@ -59,6 +60,22 @@ impl Binder {
                     err = e;
                 }
             }
+            for context in table_contexts {
```

Comment:
This may not be correct. We may only want to search for `CorrelatedInputRef` in the left tables iff depth is 1. For higher depths, its not clear if it makes sense to reference columns in left tables.

Reply:
I think this may still be correct. We may want to allow correlated input ref to left tables at depth > 1, and error if the higher subquery is not lateral.

Reply:
Btw, the current `is_correlated` impl is inadequate. It cannot detect correlation in joins.

For instance, 

```sql
    create table s (v0 int);
    create table ab (a int, b int);
    create table bc (b int, c int);
    create table t (v1 int, v2 int);
    select * from s, ab, (
      select * from bc, (
          select * from t where v1 = a and v2 = v0
      ) as t0
    );
```

Should give a "has correlated input reference" error

Reply:
Basic question: should a lateral subquery be able to have a correlated reference at depth 2?

Reply:
I think one way to solve this is to pop and merge the context if we detect a lateral join. Then we push it again.

Then we make all correlated input refs invisible for left tables if no lateral keyword is provided.

Reply:
I think the only issue with this approach is that we cannot detect `CorrelatedInputRef`...

Reply:
Actually, there are more issues with the current implementation. If lateral context is visible, then the subquery's indices are defined by the global context. However, if the lateral context is not visible, it is defined by local context. I wonder if this would lead to contradictions...